components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -332,35 +339,53 @@ private PostAuthnHandlerFlowStatus handleRequestFlow(HttpServletRequest request,
+                                username, request);
+                        // Set the property to make sure the request is a returning one.
+                        context.setProperty(FrameworkConstants.PASSWORD_PROVISION_REDIRECTION_TRIGGERED, true);
+                        return PostAuthnHandlerFlowStatus.INCOMPLETE;
                     }
-                    if (StringUtils.isEmpty(username)) {
-                        username = getUsernameFederatedUser(stepConfig, sequenceConfig, externalIdPConfigName,
-                                context, localClaimValues, externalIdPConfig);
+                    if (StringUtils.isEmpty(associatedLocalUser) && externalIdPConfig.isAssociateLocalUserEnabled()) {
+                        //TODO: Revisit this when the non-email user registration is done.
+                        if (StringUtils.isNotBlank(localClaimValues.get(EMAIL_ADDRESS_CLAIM))) {
+                            try {
+                                String emailUsername = localClaimValues.get(EMAIL_ADDRESS_CLAIM);
+                                UserRealm realm = getUserRealm(context.getTenantDomain());
+                                AbstractUserStoreManager userStoreManager =
+                                        (AbstractUserStoreManager) getUserStoreManager(context.getExternalIdP()
+                                                .getProvisioningUserStoreId(), realm, emailUsername);
+                                if (userStoreManager.isExistingUser(emailUsername)) {
+                                    User user = new User(userStoreManager.getUser(null, emailUsername));
+                                    //associate user
+                                    FrameworkUtils.getFederatedAssociationManager()
+                                            .createFederatedAssociation(user, stepConfig.getAuthenticatedIdP(),
+                                                    emailUsername);
+                                    associatedLocalUser = emailUsername;
+                                }
+                            } catch (UserStoreException e) {
+                                handleExceptions(ErrorMessages.ERROR_WHILE_CHECKING_USERNAME_EXISTENCE.getMessage(),
+                                        "error.user.existence", e);
+                            } catch (FrameworkException | FederatedAssociationManagerException e) {
+                                handleExceptions(e.getMessage(), e.getErrorCode(), e);
+                            }
+                        }
                     }
+                    String username = associatedLocalUser;
                     if (StringUtils.isNotBlank(associatedLocalUser)) {
                         // Check if the associated local account is locked.
-                        if (isAccountLocked(username, context.getTenantDomain())) {
+                        if (isAccountLocked(associatedLocalUser, context.getTenantDomain())) {
```

Comment:
since `String username = associatedLocalUser;` is there in 382 line, do we need to change username -> associatedLocalUser  here and later places

Reply:
moved the `String username = associatedLocalUser;` after the existing user conditions check.